### PR TITLE
add migrate-yaml to the upgrade script

### PR DESCRIPTION
**Pull request by prometheanfire**
Connects https://github.com/rcbops/u-suk-dev/issues/376

### CODE REVIEW

**Comment by mattt416** [comment]
@prometheanfire Yeah, I'd say make the change now since this has to be rechecked anyway.  Also, can you update `--for-testing-only` so that it doesn't need an argument?

**Comment by prometheanfire** [comment]
The default is to not require an argument.

**Comment by alextricity25** [comment]
recheck_ceph

**Comment by sigmavirus24** [comment]
Once tests pass, this looks 👍 to me. I haven't tested it yet though.

File: scripts/migrate-yaml.py [hunk 1]
```diff
@@ -34,6 +34,9 @@ def parse_args():
                         help='The path to the file with defaults.')
     parser.add_argument('--output-file', dest='output_file', required=True,
                         help='The path to the new overrides file location')
+    parser.add_argument('--for-testing-only', dest='testing',
```
Comment (hughsaunders) [inline scripts/migrate-yaml.py]: I'd prefer this option had a name that indicates its function.

Comment (prometheanfire) [inline scripts/migrate-yaml.py]: given the people that use this option I'd rather warn off improper usage than have it be something like 'new-vars-only'

**Comment by rackernexus** [comment]
Your content preview [is now ready](https://staging.developer.rackspace.com/staging.horse/build-46f122d935/docs/private-cloud/rpc/release-notes/). :bow:

**Comment by alextricity25** [comment]
:+1: I tested this and everything looks good!

File: scripts/test-upgrade.sh [hunk 2]
```diff
@@ -14,13 +14,28 @@
...
 OA_DIR="$BASE_DIR/openstack-ansible"
 RPCD_DIR="$BASE_DIR/rpcd"
 
+# TASK #0
+# Bug: https://github.com/rcbops/u-suk-dev/issues/347
+# Issue: Variables files have been reorganized, run a migration script to move
+#        them to the new location.
+"${BASE_DIR}"/scripts/migrate-yaml.py --for-testing-take-new-vars-only \
+    --defaults "${RPCD_DIR}"/etc/openstack_deploy/user_rpco_variables_defaults.yml \
+    --overrides /etc/openstack_deploy/user_extra_variables.yml \
```
Comment (alextricity25) [inline scripts/test-upgrade.sh]: Typo. s/user_extra_variables/user_extras/variables

**Pull request merged**